engine/context/src/main/java/io/deephaven/engine/context/QueryScope.java

```diff
@@ -276,14 +284,19 @@ public <T> T readParamValue(final String name, final T defaultValue) {
 
         @Override
         public <T> void putParam(final String name, final T value) {
+            if (value instanceof LivenessReferent && DynamicNode.notDynamicOrIsRefreshing(value)) {
```

Comment:
For StandaloneQueryScope: I'm not aware of another way that things can enter the scope.

For ScriptSessionQueryScope:
The expectation is that if not directly set via putParam, then it was a user from within the script session assigning a global, via executing code. Since executing code requires first opening the query scope as a liveness node, that will then be captured too.